src/main/java/com/google/sps/python/heatMap.py

```diff
@@ -0,0 +1,207 @@
+    'New Mexico': 'NM',
+    'New York': 'NY',
+    'North Carolina': 'NC',
+    'North Dakota': 'ND',
+    'Northern Mariana Islands':'MP',
+    'Ohio': 'OH',
+    'Oklahoma': 'OK',
+    'Oregon': 'OR',
+    'Pennsylvania': 'PA',
+    'Puerto Rico': 'PR',
+    'Rhode Island': 'RI',
+    'South Carolina': 'SC',
+    'South Dakota': 'SD',
+    'Tennessee': 'TN',
+    'Texas': 'TX',
+    'Utah': 'UT',
+    'Vermont': 'VT',
+    'Virgin Islands': 'VI',
+    'Virginia': 'VA',
+    'Washington': 'WA',
+    'West Virginia': 'WV',
+    'Wisconsin': 'WI',
+    'Wyoming': 'WY'
+}
+
+
+# Loads data from all three datasets above and returns a
+# python dict that maps a county's name to its coordinates
+# and a covid risk weight based on recency of outbreak
+# at that locationa and that county's population
+def loadCovidData():
+    countyToCoords = readCountyWise(coordFile)
+    popMap = loadPopMap(popFile, countyToCoords)
+    weightedCases = loadWeightedCases(caseFile, popMap)
+    i = 0
+    for county in popMap:
+        popMap[county]["weight"] = int(weightedCases[i]
+          / popMap[county]["weight"]) # Normalizes weights by population density.
+        i += 1
+    return popMap
+
+
+# Reads file to create a python dict that
+# maps county names to their coordinates.
+def readCountyWise(file):
+    countyCoords = {}
+    line_count = 0
+    f = open(file, "r")
+    for line in f:
+        if line_count > 1:  # Skip first two lines.
+            coords = {}
+            row = line.split(',')
+            key = formatCountyState(row[5], row[6])
+            # If we find a new key.
+            if len(key) > 0 and (key not in countyCoords):
+                coords["lat"] = float(row[8])
+                coords["lng"] = float(row[9])
+                countyCoords[key] = coords
+        line_count += 1
+    return countyCoords
+
+
+# Formats a county and state such as Sonoma and Washington
+# as Sonoma, WA
+def formatCountyState(county, state):
+    key = ""
+    if state in us_state_abbrev:
+        key = county + ", " + us_state_abbrev[state]
+    return key
+
+
+# Reads county-population.txt and returns python dict that matches
+# a county's name to its coordinates and relative population (population / 100)
+def loadPopMap(file, countyToCoords):
+=======
+
+
+def loadCovidData():
+    emptyLocations = [96, 193, 1863]  # Edge case locations where I know the population is 0 (ex: Princess Cruise Ship)
+    popMap = loadPopMap(popFile)  # county: population
+    weightedCases = loadWeightedCases(caseFile, emptyLocations)
+    i = 0
+    for county in popMap:
+        popMap[county] = int(weightedCases[i] / popMap[county]) # Normalizes weights by population density.
+    return popMap
+
+
+def loadPopMap(file):
+>>>>>>> e96ce87158e98b43946b4de4f76acd8635db2324
+    popMap = {}
+    line_count = 0
+    f = open(file, "r")
+    for line in f:
+<<<<<<< HEAD
+        if line_count > 1 and (line_count not in ignoreLocations):  # Skip first two lines.
```

Comment:
I'd suggest if (line_count <= 1) // skip elif (line_count in ignoreLocation) else

Just to make the conditional values simpler.

Reply:
this would simplify the if statement but then I'd have a conditional that duplicates code:
if line_count <=1:
//skip
elif line_count in ignoreLocation:
//skip
else:

is this preferred?

Reply:
I saw your latest changes, it looks better now. Marking resolved.